### DISCUSSION TITLE
Stack/Unstack Similar Photos

### DISCUSSION THREAD

**Discussion by haritvsmurali** [ente/ente#3043] [Enhancements]
Stack Photos - to group similar looking photos automatically. and also option to manually stack, unstack incase ente doesn't automatically recognize similar images. Google Photos has this feature

**Comment by stefanosweb** [comment]
Something like that: 
https://github.com/slavabarkov/tidy

**Comment by TMKI5** [comment]
I've been testing both Ente and Immich, and Immich also has this feature. It seems really useful and allows my main page to be much less crowded. There are cases I want to keep similar photos, but don't want to see 10+ in my home page.

**Reply by donateur** [reply]
I previously used Photoprism before trying Ente and it also has this feature.

I also find it annoying that Ente won't allow me to pin the "Info" detailed information pane permanently.  If this was always visible, it would be easy to flip back and forth between photos in order to determine duplicates or those which are compressed versions of existing photos.

**Comment by TrulyFlawed** [comment]
This would also be a useful feature for people who save sequences of images (such as comics) since it would allow us to collapse them into a single item and reduce clutter in our photos. I'd love to see this feature!

**Comment by jnzd** [comment]
I have my phone set to store both a raw and a jpeg photo. The gallery quickly gets cluttered with the two versions not being stacked together. It would be much cleaner if there was a stack option. Similarly if I edit a picture slightly and save it as a copy, it would be great to have the option to have it stacked with the original.

**Comment by migamic** [comment]
Besides what has already been mentioned about decluttering the gallery view, I think this feature could also help in better sorting people by the number of memories.

Currently if there are many repetitions of one picture (e.g. several takes of the same group photo, or attempts to get a nice shot of someone), they count as separate memories for a person, rather than one. When sorting people by the amount of memories, I often find that some "irrelevant" people appear high on the list, only because their photos are repeated many times.

While it might be interesting to see the total number of photos/videos that I have from a person (current behavior), I think it's more relevant to know how many *unique* memories there are (so counting repetitions as a single memory), and order the people list accordingly.

**Comment by donateur** [comment]
Hey,

It’s not what you’re asking for here, but do you realise that under the “Free up space” option in the menu there are options to compare similar images?

Unfortunately you need to use Ente photos on a desktop or large tablet to clearly be able to see differences in photos.  I wish they’d improve it so you could do a full screen comparison by flipping between images, but I guess that’s a different feature request to raise!

**Reply by juliadin** [reply]
Does Desktop even have the 'similar images' feature? I only see it on the mobile app. Using 1.7.29 on desktop via flatpak.

**Reply by donateur** [reply]
Sorry, you’re right, it only seems to be on mobile :(

**Reply by juliadin** [reply]
No harm done :) - but I hope it finds it's way over to desktop. I see how these days most media management stuff is mobile focused but the interface on mobile can't really show details yet ... or tell me which variant of an image is cropped and which is full size. I hope that the feature matures a bit but it's a very good start. 

do you happen to know where it gets it's data? is it ML based?

**Reply by laurenspriem** [reply]
The feature is ML based. But yes, it's on our roadmap to add it to desktop

**Reply by juliadin** [reply]
Thanks for that :). 😍

**Reply by migamic** [reply]
To clarify, the issue being discussed here is about grouping the images _without deleting them_. The goal is not to free up space (although improvements of that tool would also be amazing!), but to declutter the gallery.

@gweiermann recently made a more elaborate proposal of this feature, including a UI/UX mockup, in #12627
